solutions/observability/apps/transaction-sampling.md

```diff
@@ -133,6 +133,40 @@ Tail-based sampling is implemented entirely in APM Server, and will work with tr
 
 Due to [OpenTelemetry tail-based sampling limitations](../../../solutions/observability/apps/limitations.md#apm-open-telemetry-tbs) when using [tailsamplingprocessor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/tailsamplingprocessor), we recommend using APM Server tail-based sampling instead.
 
+### Tail-based sampling performance and requirements [_tail_based_sampling_performance_and_requirements]
+
+Tail-based sampling (TBS), by definition, requires storing events locally temporarily, such that they can be retrieved and forwarded once sampling decision is made.
+
+In APM Server implementation, the events are stored temporarily on disk instead of in memory for better scalability. Therefore, it requires local disk storage proportional to APM event ingestion rate, and additional memory to facilitate disk reads and writes. Insufficient [storage limit](../../../solutions/observability/apps/transaction-sampling.md#sampling-tail-storage_limit) causes sampling to be bypassed.
+
+It is recommended to use fast disks, such as NVMe SSDs, when enabling tail-based sampling. Disk throughput and I/O may become performance bottlenecks for tail-based sampling and APM event ingestion overall. Disk writes are proportional to the event ingest rate, while disk reads are proportional to both the event ingest rate and the sampling rate.
+
+To demonstrate the performance overhead and requirements, here are some reference numbers from a standalone APM Server deployed on AWS EC2 under full load, receiving APM events containing only traces. These numbers assume no backpressure from Elasticsearch and a 10% sample rate in the tail sampling policy. Please note that these figures are for reference only and may vary depending on factors such as sampling rate, average event size, and the average number of events per distributed trace.
+
+Terminology:
+
+* Event Ingestion Rate: The throughput from the APM agent to the APM Server using the Intake v2 protocol (the protocol used by Elastic APM agents), measured in events per second.
+* Event Indexing Rate: The throughput from the APM Server to Elasticsearch, measured in events per second or documents per second.
+* Memory Usage: The maximum Resident Set Size (RSS) of APM Server process observed throughout the benchmark.
+
+| APM Server version | EC2 instance size | TBS and disk configuration                     | Event ingestion rate (events/s) | Event indexing rate (events/s) | Memory usage (GB) | Disk usage (GB) |
+|--------------------|-------------------|------------------------------------------------|---------------------------------|--------------------------------|-------------------|-----------------|
+| 9.0                | c6id.2xlarge      | TBS disabled                                   | 47220                           | 4720 (100% sampling)           | 0.98              | 0               |
```

Comment:
Is this right? It seems that the events/s ingestion rate is `47220` but the event indexing rate is 4720/s (10% of the events/s?

```suggestion
| 9.0                | c6id.2xlarge      | TBS disabled                                   | 47220                           | 47220 (100% sampling)           | 0.98              | 0               |
```

Reply:
Nice catch. I fat fingered it. I'll double check all the numbers before I merge.